### PR TITLE
refactor: Footer to match NavHeader

### PR DESCRIPTION
**Pull request by timoheddes**
After implementing the NavHeader component we noticed the Footer doesn't entirely meet our standards. This PR fixes that and also renames the Footer and its files to NavFooter.

### CODE REVIEW

File: packages/libs/react-ui/src/components/index.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export type { ITooltipProps } from './Tooltip';
 export type { ITrackerCardProps, ILabelValue } from './TrackerCard/TrackerCard';
 export type { ITreeProps } from './Tree';
 export type {
-  INavFooterRootProps,
+  INavFooterContainerProps,
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/index.ts:27]: I think Root makes more sense since the subcomponent is named Root. What do you think?

Comment (timoheddes) [reply packages/libs/react-ui/src/components/index.ts:27]: Yeah that works, I'll have NavHeader match it then in this PR.

File: packages/libs/react-ui/src/components/NavFooter/NavFooterLink.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  children,
+  href,
+  target,
+}) => {
+  return (
+    <div className={linkBoxClass} data-testid="kda-footer-link-item">
+      <span
+        className={classNames(linkClass, {
+          color: 'inherit',
+          textDecorationColor: 'inherit',
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/NavFooter/NavFooterLink.tsx:23]: can these just be part of linkClass?

File: packages/libs/react-ui/src/components/NavFooter/NavFooterLink.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        })}
+      >
+        {href !== undefined ? (
+          <a
+            style={{
+              color: 'inherit',
+              textDecorationColor: 'inherit',
+            }}
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/NavFooter/NavFooterLink.tsx:31]: can we move this to a class?

Comment (eileenmguo) [reply packages/libs/react-ui/src/components/NavFooter/NavFooterLink.tsx:31]: Actually I think these styles can be removed

File: packages/libs/react-ui/src/components/NavFooter/NavFooterLink.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            href={href}
+            target={target}
+          >
+            {children}
+          </a>
+        ) : (
+          <span>{children}</span>
```
Comment (eileenmguo) [inline packages/libs/react-ui/src/components/NavFooter/NavFooterLink.tsx:38]: you can remove the extra wrapping span

File: packages/libs/react-ui/src/components/NavFooter/NavFooter.stories.tsx [hunk 5]
```diff
@@ -102,14 +103,8 @@ export const Primary: Story = {
...
-              <NavFooter.Link key={index}>
-                {item.href !== undefined ? (
-                  <a href={item.href} target={item.target}>
-                    {item.title}
-                  </a>
-                ) : (
-                  <span>{item.title}</span>
-                )}
+              <NavFooter.Link key={index} href={item.href} target={item.target}>
+                {item.title}
```
[the remark this answers is not in the document]
Comment (timoheddes) [reply packages/libs/react-ui/src/components/NavFooter/NavFooter.stories.tsx:107]: Ah ok, this is just a story though. But I'll change it!

**Review by eileenmguo — APPROVED** [review]

**Pull request merged**